consensus/src/da_member.rs

```diff
@@ -0,0 +1,289 @@
+                        if parent_state.validate_block(&p.data.deltas, &self.cur_view) {
+                            warn!("Invalid block.");
+                            return None;
+                        }
+                        let state = if let Ok(state) =
+                            parent_state.append(&p.data.deltas, &self.cur_view)
+                        {
+                            state
+                        } else {
+                            warn!("Failed to append state in high qc for proposal.");
+                            return None;
+                        };
+
+                        let leaf = DALeaf {
+                            view_number: self.cur_view,
+                            height: parent.height + 1,
+                            justify_qc: self.high_qc.clone(),
+                            parent_commitment: parent.commit(),
+                            deltas: p.data.deltas.clone(),
+                            state: Left(state),
+                            rejected: Vec::new(),
+                            timestamp: time::OffsetDateTime::now_utc().unix_timestamp_nanos(),
+                            proposer_id: sender.to_bytes(),
+                        };
+
+                        let block_commitment = p.data.deltas.commit();
+                        if !view_leader_key.validate(&p.signature, block_commitment.as_ref()) {
+                            warn!(?p.signature, "Could not verify proposal.");
+                            continue;
+                        }
+
+                        let consensus = self.consensus.read().await;
+
+                        // Liveness check.
+                        let liveness_check = self.high_qc.view_number() > consensus.locked_view + 2;
+
+                        // Safety check.
+                        // Check if the proposal extends from the locked leaf.
+                        let outcome = consensus.visit_leaf_ancestors(
+                            parent.view_number,
+                            Terminator::Inclusive(consensus.locked_view),
+                            false,
+                            |leaf| {
+                                // if leaf view no == locked view no then we're done, report success by
+                                // returning true
+                                leaf.view_number != consensus.locked_view
+                            },
+                        );
+                        let safety_check = outcome.is_ok();
+                        if let Err(e) = outcome {
+                            self.api.send_view_error(self.cur_view, Arc::new(e)).await;
+                        }
+
+                        // Skip if both saftey and liveness checks fail.
+                        if !safety_check && !liveness_check {
+                            warn!("Failed safety check and liveness check");
```

Comment:
Are safety and liveness checks necessary here? My (potentially incorrect) understanding is that we check for structural integrity and then immediately vote yes. And then these liveness and safety checks happen later when we are a replica validating the CommitmentProposal.

Reply:
+1, At the very least replicas not on the committee will have to do this same check

Reply:
I wonder if we need the liveness check to be +2 or +1. +2 was added to make the chernoff bounds work when we were doing VRF on consensus (not DA). Since we're not doing that anymore, when voting on the CommitmentProposal, I think we can change the liveness check back to +1.

Reply:
Thanks! I've removed the safety and liveness checks here. Also fixed the liveness check for the validating replica.